src/components/registry-of-geography/registry-of-geography.service.ts

```diff
@@ -0,0 +1,246 @@
+    );
+
+    const secureProps = [
+      {
+        key: 'name',
+        value: input.name,
+        addToAdminSg: true,
+        addToWriterSg: false,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+        label: 'FieldZoneName',
+      },
+      {
+        key: 'registryId',
+        value: input.registryId,
+        addToAdminSg: true,
+        addToWriterSg: false,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+        label: 'RegistryOfGeographyId',
+      },
+    ];
+
+    try {
+      const query = this.db
+        .query()
+        .call(matchRequestingUser, session)
+        .match([
+          node('rootUser', 'User', {
+            id: this.config.rootAdmin.id,
+          }),
+        ])
+        .call(createBaseNode, 'RegistryOfGeography', secureProps)
+        .return('node.id as id');
+
+      const result = await query.first();
+      if (!result) {
+        throw new ServerException('Failed to create registry of geography');
+      }
+
+      this.logger.info(`registry of geography created`, { id: result.id });
+
+      return await this.readOne(result.id, session);
+    } catch (err) {
+      this.logger.error('Could not create registry of geography for user', {
+        exception: err,
+        userId: session.userId,
+      });
+      throw new ServerException('Could not create registry of geography');
+    }
+  }
+
+  async readOne(id: string, session: ISession): Promise<RegistryOfGeography> {
+    this.logger.info('readOne', { id, userId: session.userId });
+
+    if (!id) {
+      throw new NotFoundException('no id given');
+    }
+
+    if (!session.userId) {
+      session.userId = this.config.anonUser.id;
+    }
+
+    const readRegistryOfGeography = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'RegistryOfGeography', { id })])
+      .call(matchPermList, 'requestingUser')
+      .call(matchPropList, 'permList')
+      .return('propList, permList, node')
+      .asResult<StandardReadResult<DbPropsOfDto<RegistryOfGeography>>>();
+
+    const result = await readRegistryOfGeography.first();
+
+    if (!result) {
+      throw new NotFoundException('RegistryOfGeography.id', 'id');
+    }
+
+    const secured = parseSecuredProperties(
+      result.propList,
+      result.permList,
+      this.securedProperties
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...secured,
+    };
+  }
+
+  async update(
+    input: UpdateRegistryOfGeography,
+    session: ISession
+  ): Promise<RegistryOfGeography> {
+    const RegistryOfGeography = await this.readOne(input.id, session);
+
+    return await this.db.sgUpdateProperties({
+      session,
+      object: RegistryOfGeography,
+      props: ['name', 'registryId'],
+      changes: input,
+      nodevar: 'registryOfGeography',
+    });
+  }
+
+  async delete(id: string, session: ISession): Promise<void> {
```

Comment:
deletes are being refacored. this can be skipped for now. we won't use ACLs in the future.